Correct implementation of create_advanced_sequence to support repeated capabilities

ni-erikcrank opened the pull request:
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- ~~[ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?
Adds support for repeated capabilities to create_advanced_sequence method. A previous attempt to add this support was incomplete.

### List issues fixed by this Pull Request below, if any.

### What testing has been done?
Verified calls to create_advanced_sequence using repeated capabilities succeed without error. New system tests will be added to test each method that now supports repeated capabilities in a different PR #1624.

sbethur commented:
ok to test

sbethur commented:
Title says fixing create_advanced_sequence() but diff includes new system tests. Can you please split this into 2 PRs.

ni-erikcrank commented:
> 
> 
> Title says fixing create_advanced_sequence() but diff includes new system tests. Can you please split this into 2 PRs.

Removed the system tests and updated description.
